pallets/subtensor/src/swap/swap_coldkey.rs

```diff
@@ -59,28 +59,35 @@ impl<T: Config> Pallet<T> {
             Error::<T>::NotEnoughBalanceToPaySwapColdKey
         );
 
-        // 6. Remove and burn the swap cost from the old coldkey's account
+        // 6. Swap identity if the old coldkey has one.
+        if Identities::<T>::contains_key(&old_coldkey)
+            && !Identities::<T>::contains_key(new_coldkey)
+        {
+            Self::swap_delegate_identity_coldkey(&old_coldkey, new_coldkey)?;
```

Comment:
Please follow the existing pattern in the  swap function i.e. we do not use separate functions , as it affects readability. Perform all the steps inline